andela/knights-bn-backend
# #170947681 Allow travel admin to create accommodation facilities

#### What does this PR do?
- Add functionality of creating accommodation facilities 
#### Description of Task to be completed?
- only travel admin will be able to create and edit accommodation facilities by providing `location name`,`streetNumber`,`number of rooms`,`type of rooms`and `building image`
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn   backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-create-accommodation-170947681` 
- run `npm install` to install dependencies
- check if you have PostgresQL database, if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run dropTables` to clean your database
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have
- login by sending POST request to this url ``http://localhost:4000/api/v1/auth/login with the credentials I mentioned below
```
{
 "email":"traveladmin@gmail.com",
   "password": "Niyonkuru@1"
}
```
- create an accommodation facility by making POST request to this url `http://localhost:4000/api/v1/create/accommodation` along with this data 
```
{
 "accommodationName":"Hill top",
 "locationName": "Chicago",
 "streetNumber": "104st ",
   "numberOfRooms": 4,
   "description":"it has wifi and big secreen TV",
    "availableRooms": [
      {
        "roomName": "double room",
        "roomType": "king",
        "price":"$50/night"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$53/night"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$54/night"
      }
      ,
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$55/night"
      }
    ]
  }
```

- to view all the accommodations sent GET request to this URL `http://localhost:4000/api/v1/view/accommodations`
- to view specific accommodation make a GET request to this URL `http://localhost:4000/api/v1/view/accommodation/1` the number one `(1)` you see here is  accommodation `ID`, you can replace it with a another accommodation `ID`  you want to view
- to upload image of a specific accommodation use your postman as a form and name one field `imageOfBuilding` and give it file type in order to upload accommodation or center image, after uploading your image send PATCH request to this url`http://localhost:4000/api/v1/upload/accommodation/1`  `1` is ID of this accommodation we want to give an image
- if you want to edit a specific accommodation information make a PATCH request to this url `http://localhost:4000/api/v1/edit/accommodation/1`,  you can use the data provided below or write your own
```
{

    "availableRooms": [
      {
        "roomName": "double room",
        "roomType": "king",
        "price":"$50/night",
        "description":"it has wifi and big secreen TV"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$53/night",
        "description":"it has wifi and big secreen TV"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$54/night",
        "description":"it has wifi and big secreen TV"
      }
      ,
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$55/night",
        "description":"it has wifi and big secreen TV"
      }
    ]
  }
``` 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947681](https://www.pivotaltracker.com/story/show/170947681)
#### Screenshots

#### Create Accommodation
<img width="890" alt="Screen Shot 2020-03-09 at 18 37 14" src="https://user-images.githubusercontent.com/36619897/76241361-1c8cce00-6235-11ea-91e8-db84bab333dd.png">


#### Upload image
<img width="808" alt="Screen Shot 2020-03-07 at 17 50 02" src="https://user-images.githubusercontent.com/36619897/76146732-fda80380-609d-11ea-9c5e-d19f651041d1.png">

#### View all accommodation
<img width="930" alt="Screen Shot 2020-03-09 at 18 42 38" src="https://user-images.githubusercontent.com/36619897/76241767-c9ffe180-6235-11ea-8ea0-13793bc564eb.png">

#### Edit accommodation
<img width="1038" alt="Screen Shot 2020-03-09 at 18 40 42" src="https://user-images.githubusercontent.com/36619897/76241623-8efdae00-6235-11ea-8ca0-87dafab95698.png">
